src/libs/actions/Policy/Policy.ts

```diff
@@ -3740,6 +3749,634 @@ function getAdminPoliciesConnectedToNetSuite(): Policy[] {
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                shouldShowAutoApprovalOptions: !enabled,
+                pendingFields: {
+                    shouldShowAutoApprovalOptions: null,
+                    ...autoApprovalCleanupValues,
+                },
+                errorFields: {
+                    shouldShowAutoApprovalOptions: ErrorUtils.getMicroSecondOnyxErrorWithTranslationKey('common.genericErrorMessage'),
+                },
+            },
+        },
+    ];
+
+    const parameters: EnablePolicyAutoApprovalOptionsParams = {
+        enabled,
+        policyID,
+    };
+
+    API.write(WRITE_COMMANDS.ENABLE_POLICY_AUTO_APPROVAL_OPTIONS, parameters, {
+        optimisticData,
+        successData,
+        failureData,
+    });
+}
+
+/**
+ * Call the API to set the limit for auto-payments in the given policy
+ * @param policyID - id of the policy to apply the limit to
+ * @param limit - max amount for auto-payment for the reports in the given policy
+ */
+function setPolicyAutoReimbursementLimit(policyID: string, limit: string) {
+    const policy = getPolicy(policyID);
+    const fallbackLimit = limit === '' ? '0' : limit;
+    const parsedLimit = CurrencyUtils.convertToBackendAmount(parseFloat(fallbackLimit));
+
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.FORMS.RULES_AUTO_PAY_REPORTS_UNDER_MODAL_FORM,
+            value: {
+                maxExpenseAutoPayAmount: limit,
+            },
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                autoReimbursement: {
+                    limit: parsedLimit,
+                },
+                pendingFields: {
+                    autoReimbursement: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                },
+            },
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.FORMS.RULES_AUTO_PAY_REPORTS_UNDER_MODAL_FORM,
+            value: {
+                maxExpenseAutoPayAmount: limit,
+            },
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                autoReimbursement: {
+                    limit: parsedLimit,
+                },
+                pendingFields: {
+                    autoReimbursement: null,
+                },
+                errorFields: null,
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                autoReimbursement: policy?.autoReimbursement,
```

Comment:
Why don't we revert to policy.autoReimbursement.limit 
```
autoReimbursement: {
          limit: policy?.autoReimbursement?.limit ?? policy?.autoReimbursement
},
```

Reply:
why do you want this kind of fallback, is it a deprecated value?

Reply:
@BrtqKr In the new change in this PR, the value will be saved in policy?.autoReimbursement?.limit. 
1. Set new value --> the new value is saved in policy?.autoReimbursement?.limit
2. Update to another value --> Failure
3. Then, we need to revert to policy?.autoReimbursement?.limit

I think we only should revert to policy?.autoReimbursement if policy?.autoReimbursement?.limit is undef